internal/cache/controller_obj_cache.go

```diff
@@ -2554,6 +2541,51 @@ func validateAndConfigureSeGroup(client *clients.AviClient) bool {
 		return false
 	}
 
+	// Indulge in Marker based SEGroup searching only when user input is empty.
+	// SE Group selection priority
+	// 1. User input
+	seGroupToUse := lib.GetSEGNameEnv()
+
+	// 2. Marker based
+	if seGroupToUse == "" {
+		for _, elem := range elems {
+			seg := models.ServiceEngineGroup{}
+			err = json.Unmarshal(elem, &seg)
+			if err != nil {
+				utils.AviLog.Warnf("Failed to unmarshal data, err: %v", err)
+				continue
+			}
+
+			clusterName := lib.GetClusterName()
+			if lib.GetAdvancedL4() {
+				clusterName = lib.GetClusterID()
+			}
+
+			if len(seg.Markers) == 1 &&
```

Comment:
You are altering the behaviour of regular AKO flows here. It looks like you are assuming that if the se group is labeled with cluster name in the `marker` interface, we will take that SE group into consideration. This is in conflict to the labelling done on the SE group which is a route table label instead of an object label.